src/coreclr/jit/optimizer.cpp

```diff
@@ -9657,9 +9701,15 @@ GenTree* OptBoolsDsc::optIsBoolComp(OptTestInfo* pOptTest)
     assert(pOptTest->testTree->gtOper == GT_JTRUE || pOptTest->testTree->gtOper == GT_RETURN);
     GenTree* cond = pOptTest->testTree->AsOp()->gtOp1;
 
-    // The condition must be "!= 0" or "== 0"
+    // The condition must be "!= 0" or "== 0" or >=0 or <0
+
+    if (!cond->OperIs(GT_EQ, GT_NE, GT_LT, GT_GE))
+    {
+        return nullptr;
+    }
```

Comment:
Will the transformations handle unsigned comparisons correctly? Should this instead be:
```suggestion
    if (!cond->OperIs(GT_EQ, GT_NE) && (!cond->OperIs(GT_LT, GT_GE) || cond->IsUnsigned()))
    {
        return nullptr;
    }
```
?